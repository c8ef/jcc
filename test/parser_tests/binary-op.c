int foo(void) {
  int x = 12 + 30;
  int y = 1 + 2 * 3;
  int z = 10 / 5 - 2;

  x += 2;
  y -= 3;
  z /= 1;
  x *= 2;
  x == y;
}
