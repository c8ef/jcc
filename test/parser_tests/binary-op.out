FunctionDecl: foo
  Args[0]
  CompoundStatement
    DeclStatement
      VarDecl: x
        BinaryExpr(+):
          IntergerLiteral: 12
          IntergerLiteral: 30
    DeclStatement
      VarDecl: y
        BinaryExpr(+):
          IntergerLiteral: 1
          BinaryExpr(*):
            IntergerLiteral: 2
            IntergerLiteral: 3
    DeclStatement
      VarDecl: z
        BinaryExpr(-):
          BinaryExpr(/):
            IntergerLiteral: 10
            IntergerLiteral: 5
          IntergerLiteral: 2
    ExprStatement
      BinaryExpr(+=):
        DeclRefExpr: x
        IntergerLiteral: 2
    ExprStatement
      BinaryExpr(-=):
        DeclRefExpr: y
        IntergerLiteral: 3
    ExprStatement
      BinaryExpr(/=):
        DeclRefExpr: z
        IntergerLiteral: 1
    ExprStatement
      BinaryExpr(*=):
        DeclRefExpr: x
        IntergerLiteral: 2
    ExprStatement
      BinaryExpr(==):
        DeclRefExpr: x
        DeclRefExpr: y
